src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -300,24 +290,41 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     {
         if (!_KeyBindingMapCache)
         {
-            // populate _KeyBindingMapCache
-            std::unordered_map<KeyChord, Model::Command, KeyChordHash, KeyChordEquality> keyBindingsMap;
-            std::unordered_set<ActionMapKeyChord> unboundKeys;
-            _PopulateKeyBindingMapWithStandardCommands(keyBindingsMap, unboundKeys);
-
-            _KeyBindingMapCache = single_threaded_map<KeyChord, Model::Command>(std::move(keyBindingsMap));
+            _RefreshKeyBindingCaches();
         }
         return _KeyBindingMapCache.GetView();
     }
 
+    void ActionMap::_RefreshKeyBindingCaches()
+    {
+        std::unordered_map<KeyChord, Model::Command, KeyChordHash, KeyChordEquality> keyBindingsMap;
+        std::unordered_map<KeyChord, Model::Command, KeyChordHash, KeyChordEquality> globalHotkeys;
+        std::unordered_set<Control::KeyChord, KeyChordHash, KeyChordEquality> unboundKeys;
+
+        _PopulateKeyBindingMapWithStandardCommands(keyBindingsMap, unboundKeys);
+
+        for (const auto& [keys, cmd] : keyBindingsMap)
+        {
+            // Only populate GlobalHotkeys with actions whose
+            // ShortcutAction is GlobalSummon or QuakeMode
+            if (cmd.ActionAndArgs().Action() == ShortcutAction::GlobalSummon || cmd.ActionAndArgs().Action() == ShortcutAction::QuakeMode)
+            {
+                globalHotkeys.emplace(keys, cmd);
+            }
+        }
+
+        _KeyBindingMapCache = single_threaded_map<KeyChord, Model::Command>(std::move(keyBindingsMap));
+        _GlobalHotkeysCache = single_threaded_map<KeyChord, Model::Command>(std::move(globalHotkeys));
+    }

```

Comment:
I'm actually shocked that this works -- does it just infer the template args for Hash and Equality? Whoa.

Reply:
Actually they're all inferred from the argument. I've removed the template arguments.